pkg/initializers/types/types.py

```diff
@@ -0,0 +1,17 @@
+from dataclasses import dataclass
+from typing import Optional
+
+
+# Configuration for the HuggingFace dataset initializer.
+# TODO (andreyvelich): Discuss how to keep these configurations is sync with Kubeflow SDK types.
+@dataclass
+class HuggingFaceDatasetInitializer:
+    storage_uri: str
+    access_token: Optional[str] = None
+
+
+# Configuration for the HuggingFace model initializer.
+@dataclass
+class HuggingFaceModelInitializer:
+    storage_uri: str
+    access_token: Optional[str] = None
```

Comment:
I decoupled these configs and [mount path constants](https://github.com/kubeflow/trainer/blob/972bfea819dc5732ab7ad8f72ab238ce60e63a45/pkg/initializers/utils/utils.py#L11) from the Kubeflow SDK.
I guess, this is locally correct given that our system components should not depend on client SDK.

In the future, we should talk about how to keep these configs in sync between client and server side.

cc @seanlaii @astefanutti @kubeflow/wg-training-leads @Electronic-Waste